cargo test fails on current main branch due to default engine doctest failures

Running `cargo test` (as recommended in `CONTRIBUTING.md`) as at commit `cfedb382a41babc3a334bcd26230b3e1e4a2ea32` fails with:
```
   Doc-tests delta_kernel_default_engine

running 3 tests
test default-engine/src/storage.rs - storage::store_from_url (line 48) ... FAILED
test default-engine/src/lib.rs - DefaultEngineBuilder (line 105) - compile ... FAILED
test default-engine/src/storage.rs - storage::store_from_url_opts (line 71) ... FAILED

failures:

---- default-engine/src/storage.rs - storage::store_from_url (line 48) stdout ----
error[E0433]: cannot find `default` in `engine`
  --> default-engine/src/storage.rs:50:27
   |
50 | use delta_kernel::engine::default::storage::store_from_url;
   |                           ^^^^^^^ could not find `default` in `engine`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0433`.
Couldn't compile the test.
---- default-engine/src/lib.rs - DefaultEngineBuilder (line 105) stdout ----
error[E0433]: cannot find `default` in `engine`
   --> default-engine/src/lib.rs:108:27
    |
108 | use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
    |                           ^^^^^^^ could not find `default` in `engine`

error[E0432]: unresolved import `delta_kernel::engine::default`
   --> default-engine/src/lib.rs:107:27
    |
107 | use delta_kernel::engine::default::DefaultEngineBuilder;
    |                           ^^^^^^^ could not find `default` in `engine`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
Couldn't compile the test.
---- default-engine/src/storage.rs - storage::store_from_url_opts (line 71) stdout ----
error[E0433]: cannot find `default` in `engine`
  --> default-engine/src/storage.rs:74:27
   |
74 | use delta_kernel::engine::default::storage::store_from_url_opts;
   |                           ^^^^^^^ could not find `default` in `engine`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0433`.
Couldn't compile the test.

failures:
    default-engine/src/lib.rs - DefaultEngineBuilder (line 105)
    default-engine/src/storage.rs - storage::store_from_url (line 48)
    default-engine/src/storage.rs - storage::store_from_url_opts (line 71)
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -227,6 +227,8 @@ jobs:
       - uses: taiki-e/install-action@98ec31d284eb962f41c14065e9391a955aa810cf # nextest
       - name: test
         run: cargo nextest run --locked --workspace --all-features -E 'not test(read_table_version_hdfs) and not test(invalid_handle_code)'
+      - name: doc tests
+        run: cargo test --locked --workspace --all-features --doc
       - name: trybuild tests
         if: steps.filter.outputs.ffi == 'true'
         run: cargo test --locked --package delta_kernel_ffi --features internal-api -- invalid_handle_code

--- a/default-engine/src/lib.rs
+++ b/default-engine/src/lib.rs
@@ -104,8 +104,8 @@ pub struct DefaultEngine<E: TaskExecutor> {
 ///
 /// ```no_run
 /// # use std::sync::Arc;
-/// # use delta_kernel::engine::default::DefaultEngineBuilder;
-/// # use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
+/// # use delta_kernel_default_engine::DefaultEngineBuilder;
+/// # use delta_kernel_default_engine::executor::tokio::TokioBackgroundExecutor;
 /// # use delta_kernel::object_store::local::LocalFileSystem;
 /// // Build a DefaultEngine with default executor
 /// let engine = DefaultEngineBuilder::new(Arc::new(LocalFileSystem::new()))

--- a/default-engine/src/storage.rs
+++ b/default-engine/src/storage.rs
@@ -47,7 +47,7 @@ pub fn insert_url_handler(
 ///
 /// ```rust
 /// # use url::Url;
-/// # use delta_kernel::engine::default::storage::store_from_url;
+/// # use delta_kernel_default_engine::storage::store_from_url;
 /// # use delta_kernel::DeltaResult;
 /// # fn example() -> DeltaResult<()> {
 /// let url = Url::parse("file:///path/to/table")?;
@@ -71,7 +71,7 @@ pub fn store_from_url(url: &Url) -> delta_kernel::DeltaResult<Arc<dyn ObjectStor
 /// ```rust
 /// # use url::Url;
 /// # use std::collections::HashMap;
-/// # use delta_kernel::engine::default::storage::store_from_url_opts;
+/// # use delta_kernel_default_engine::storage::store_from_url_opts;
 /// # use delta_kernel::DeltaResult;
 /// # fn example() -> DeltaResult<()> {
 /// let url = Url::parse("s3://my-bucket/path/to/table")?;

